src/Components/Boilerplates/BoilerplatesIndex.jsx

```diff
@@ -25,11 +25,42 @@ export default function BoilerplatesIndex() {
     isError,
     isLoading,
     error,
-    _refetch,
+    refetch: refetchBoilerplates,
   } = useQuery("getBoilerplates", () =>
     BoilerplatesService.getAllBoilerplates(organizationClient)
   );
 
+  const { mutate: updateBoilerplate } = useMutation(
+    (boilerplateFields) =>
+      BoilerplatesService.updateBoilerplate(
+        organizationClient,
+        boilerplateFields.id,
+        boilerplateFields
+      ),
+    {
+      onSuccess() {
+        refetchBoilerplates();
+      },
+    }
+  );
+
+  const handleDropdownMiniAction = async ({ option, boilerplate }) => {
+    try {
+      switch (option.value) {
+        case "REMOVE_FROM_ARCHIVED":
+          updateBoilerplate({ id: boilerplate.id, archived: false });
+          break;
+        case "MARK_AS_ARCHIVED":
+          updateBoilerplate({ id: boilerplate.id, archived: true });
+          break;
+        default:
+          throw new Error(`Unexpected option given ${option.value}!`);
+      }
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
Is there a particular reason we wrap this in a try/catch? I think it would be fine without.